### DISCUSSION TITLE
Error during compiling (Ubuntu 14.04 x64)

### DISCUSSION THREAD

**Discussion by DVL333** [eteran/edb-debugger#826] [Q&A] [answered]
Trying to install edb-debugger on Ubuntu 14.04 x64, according to document: https://github.com/eteran/edb-debugger/wiki/Compiling-(Ubuntu)
Execute command `cmake ..` and have error: "Your g++ version is too old.  At least 7.0 is required." 
But I have gcc version 9.4.0 (checked it with command `gcc --version`)
Where can a trouble be in ? Please help.

**Comment by eteran** [comment, marked as answer]
Could you please share the full output of the cmake command?

Sometimes it will say what it thinks the versions of things are and may offer some clues as to what is going on.

**Reply by DVL333** [reply]
@eteran 
I solved the case by myself - your question was a clue. :) I saw full output of `cmake` and noticed, that there was the mention of gcc version, that I had before. Indeed, first time when I ran `cmake` I hadn't proper gcc version and installed it. Then ran `cmake` again and took the same error. 
So solution is to delete all files in "build" dir and run `cmake ..` again. )

**Reply by eteran** [reply]
Great, yeah, cmake will cache the compiler version and keep it the first one it detects until you reset the build directory.

Glad you figured it out!

**Comment by 10110111** [comment]
> checked it with command `gcc --version`

You check the C compiler, while the build system complains about the C++ one. Please make sure that `g++ --version` prints the expected version. Note that on Ubuntu, g++ and gcc are provided by different packages.
